<a href="https://colab.research.google.com/github/ClaudioRo81/Diplomado-Tecnicas-Estadisticas-y-Mineria-de-Datos-Notebooks-Modulo-2/blob/main/Notebook_3.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Notebook 3: Estadística inferencial
Claudio Rojas Duran.

**Contenido:**
## Estimadores
- Definición de estimador.
- Método de momentos.
- Estimación por máxima verosimilitud (EMV).
- Ejemplos con Python.
- Interpretación de resultados.

## Propiedades de los estimadores
- Insesgadez.
- Consistencia.
- Eficiencia.

## Intervalos de confianza

- Intervalo de confianza para una media.
- Intervalo de confianza para una proporción.
- Uso de distribución normal o t de Student.
- Interpretación del intervalo.

---
# Fundamentos de la estadística inferencial

Buscamos hacer inferencias de una población hacia muestras (conjunto de datos aleatorios y representativos).

Sea $(X_{1},...,X_{n})$ una muestra de variables aleatorias independientes e idénticamente distribuidas (v.a.i.i.d) con una distribución dada.

*   **Características fijas de la población:**
    *   Media poblacional: $\mu$
    *   Varianza poblacional: $\sigma^2$
*   **Características de la muestra:**
    *   Media muestral: $\overline{X}_{n}=\frac{\sum x_{i}}{n}$
    *   Varianza muestral: $S_{n}^{2}=\frac{\sum(X_{i}-\overline{X})^{2}}{n-1}$

Queremos conocer el "mejor parámetro" del conjunto de datos.



## Estimadores

Un estimador es una función de la muestra aleatoria $\hat{\theta}$ que se usa para aproximar un parámetro desconocido de la población.
*   Ejemplo: $\overline{X}$ se usa para estimar $\mu$.
*   Ejemplo: $S^{2}$ se usa para estimar $\sigma^{2}$.

Si $(X_{1},...,X_{n})$ es una muestra aleatoria, $\hat{\theta}=T(X_{1},...,X_{n})$ busca aproximar o estimar a $\theta$ (por ejemplo, $T(X_{1},X_{2})=\frac{X_{1}+X_{2}}{2}$ o $T(X_{1},X_{2})=3X_{1}+X_{2}$).

El espacio parametral se define como $\Theta = \{ \theta \mid \theta \text{ es parámetro de la distribución} \}$.






## Propiedades de los estimadores
1.  **Insesgados:** $\mathbb{E}[\hat{\theta}]=\theta$.
    
    Significa que el estimador no tiene un sesgo sistemático (no tiende a inflar ni a reducir el valor)
2.  **Consistente:** $\hat{\theta}_{n} \Rightarrow_{n\rightarrow\infty} \theta$.

    Garantiza que a medida que el tamaño de tu muestra crece (tiende a infinito), el estimador se acercará cada vez más al valor verdadero del parámetro hasta coincidir con él.
3.  **Eficientes:** El que tenga menor varianza.

    Si comparamos varios estimadores el mas eficiente seria el que genere menor ruido es decir, el que tenga menos varianza, pues asi sus estimaciones estarian mas concentradas al valor real
4.  **Suficientes:** Contengan toda la información disponible.

    Significa que el estimador resume toda la informacion util que los datos aportan
5.  **Completos:** Existe $\hat{\theta}_{1}$ insesgado con menor varianza.
    
    Un estimador completo nos garantizaria tener el mejor estimador para la muestra observada.


#Métodos para Construir estimadores





## 1. Método de momentos
Consiste en igualar los momentos poblacionales con los momentos muestrales.

Sabemos que si $X$ es una variable aleatoria con $\mathbb{E}[X]=\mu$, entonces $\mathbb{E}[X^{n}]$ es el n-ésimo momento de $X$. Los momentos muestrales son $m_{n}=\frac{1}{n}\sum_{i=1}^{n}{X_{i}}^{n}$.


$\color{blue}{\text{Ejecicio estimador por Momentos v.a gamma}}$

**Ejemplo:** Hallar los estimadores por el método de momentos para $(X_{1},...,X_{n})$ una muestra aleatoria de una población con distribución $\text{Gamma}(\alpha,\beta)$. Se busca encontrar los estimadores $\hat{\alpha}$ y $\hat{\beta}$.

Sabemos que si $X \sim \text{Gamma}(\alpha, \beta)$, entonces $\mathbb{E}[X] = \frac{\alpha}{\beta}$ y $\text{Var}(X) = \frac{\alpha}{\beta^2}$. Además, sabemos que $\text{Var}(X)=\mathbb{E}[X^{2}]-(\mathbb{E}[X])^{2}$.
$$ \mathbb{E}[X^{2}] = \text{Var}(X) + (\mathbb{E}[X])^{2} = \frac{\alpha}{\beta^{2}} + \left(\frac{\alpha}{\beta}\right)^{2} = \frac{\alpha}{\beta^{2}} + \frac{\alpha^{2}}{\beta^{2}} $$

Igualamos con los momentos muestrales:
*   Para el primer momento: $\mathbb{E}[X] = \frac{1}{n}\sum X_{i} \Rightarrow \frac{\alpha}{\beta} = \overline{X} \Rightarrow \beta = \frac{\alpha}{\overline{X}}$.
*   Para el segundo momento: $\mathbb{E}[X^{2}] = \frac{1}{n}\sum X_{i}^{2}$.

Sustituyendo $\beta$:
$$ \mathbb{E}[X^{2}] = \frac{\alpha}{\beta^{2}} + \frac{\alpha^{2}}{\beta^{2}} = \frac{\alpha}{(\frac{\alpha}{\overline{X}})^{2}} + \frac{\alpha^{2}}{(\frac{\alpha}{\overline{X}})^{2}} = \frac{\overline{X}^{2}}{\alpha} + \overline{X}^{2} = \overline{X}^{2}\left(\frac{1}{\alpha} + 1\right) $$

Igualando al momento muestral $m_2$:
$$ \overline{X}^{2}\left(\frac{1}{\alpha} + 1\right) = \frac{1}{n}\sum X_{i}^{2} = m_{2} $$
$$ \left(\frac{1}{\alpha} + 1\right) = \frac{m_{2}}{\overline{X}^{2}} \Rightarrow \frac{1}{\alpha} = \frac{m_{2}}{\overline{X}^{2}} - 1 = \frac{m_{2}-\overline{X}^{2}}{\overline{X}^{2}} $$

Despejando $\alpha$:
$$ \alpha = \frac{\overline{X}^{2}}{m_{2}-\overline{X}^{2}} = \frac{\overline{X}^{2}}{\frac{1}{n}\sum(X_{i}-\overline{X})^{2}} = \frac{n\overline{X}^{2}}{\sum(X_{i}-\overline{X})^{2}} $$

Finalmente, los estimadores que dependen de $X_i$ son:
*   $\hat{\alpha} = \frac{n\overline{X}^{2}}{\sum(X_{i}-\overline{X})^{2}}$
*   $\hat{\beta} = \frac{n\overline{X}}{\sum(X_{i}-\overline{X})^{2}}$

**Conclusión :** El método de momentos nos permitió expresar los parámetros teóricos $\alpha$ y $\beta$ (que definen la forma y escala de la distribución Gamma) en términos de la media y la varianza de nuestra muestra. Esto significa que, en la práctica, con solo calcular el promedio y la dispersión empírica de nuestros datos, podemos ajustar una curva teórica Gamma.


$\color{blue}{\text{Ejercicio estimador por Momentos v.a Normal}}$

Hallar los estimadores por el método de momentos para $(X_{1},...,X_{n})$ una muestra aleatoria de una población con distribución $\mathcal{N}(\mu, \sigma^2)$. Se busca encontrar los estimadores $\hat{\mu}$ y $\hat{\sigma}^2$.Sabemos que si $X \sim \mathcal{N}(\mu, \sigma^2)$, entonces $\mathbb{E}[X] = \mu$ y $\text{Var}(X) = \sigma^2$. Además, sabemos que $\text{Var}(X)=\mathbb{E}[X^{2}]-(\mathbb{E}[X])^{2}$.$$\mathbb{E}[X^{2}] = \text{Var}(X) + (\mathbb{E}[X])^{2} = \sigma^2 + \mu^2$$Igualamos con los momentos muestrales:Para el primer momento: $\mathbb{E}[X] = \frac{1}{n}\sum X_{i} \Rightarrow \mu = \overline{X}$.Para el segundo momento: $\mathbb{E}[X^{2}] = \frac{1}{n}\sum X_{i}^{2} = m_2$.Sustituyendo $\mu$ en la ecuación del segundo momento:$$m_2 = \sigma^2 + \overline{X}^2$$Despejando $\sigma^2$:$$\sigma^2 = m_2 - \overline{X}^2 = \frac{1}{n}\sum X_{i}^{2} - \overline{X}^2$$Recordando la propiedad algebraica de las sumatorias, esta expresión es exactamente la definición de la varianza muestral (sin corregir):$$\sigma^2 = \frac{1}{n}\sum(X_{i} - \overline{X})^2$$Finalmente, los estimadores que dependen de $X_i$ son:

$\hat{\mu} = \overline{X}$$\hat{\sigma}^2 = \frac{1}{n}\sum(X_{i} - \overline{X})^2$

**Conclusión**: Al aplicar el método de momentos a una distribución Normal, descubrimos que $\mu$ se estima simplemente con la media de la muestra, y el parámetro de dispersión $\sigma^2$ se estima con la varianza poblacional de la muestra.

$\color{blue}{\text{Ejercicio estimador por Momentos v.a Uniforme}}$

Hallar los estimadores por el método de momentos para $(X_{1},...,X_{n})$ una muestra aleatoria de una población con distribución $\text{Unif}(a, b)$ continua.

Se busca encontrar los estimadores $\hat{a}$ (límite inferior) y $\hat{b}$ (límite superior).

Sabemos que si $X \sim \text{Unif}(a, b)$, entonces

 $\mathbb{E}[X] = \frac{a+b}{2}$ y $\text{Var}(X) = \frac{(b-a)^2}{12}$.

 Para facilitar el álgebra, igualaremos directamente el primer momento (media) y el momento central (varianza) con sus contrapartes muestrales.

 Igualamos con los estadísticos muestrales.

 Para el primer momento:

 $\mathbb{E}[X] = \overline{X} \Rightarrow \frac{a+b}{2} = \overline{X} \Rightarrow a+b = 2\overline{X}$.

 Para la varianza: $\text{Var}(X) = \frac{1}{n}\sum(X_{i} - \overline{X})^2 = \hat{S}^2$.

 Sustituyendo la varianza teórica con la empírica:$$\frac{(b-a)^2}{12} = \hat{S}^2 \Rightarrow (b-a)^2 = 12\hat{S}^2 \Rightarrow b-a = \sqrt{12\hat{S}^2}$$Ahora tenemos un sistema de ecuaciones lineales con dos incógnitas ($a$ y $b$):

 $b + a = 2\overline{X}$$b - a = \sqrt{12\hat{S}^2} = 2\sqrt{3\hat{S}^2}$ Sumando la ecuación (1) y la ecuación (2) eliminamos $a$:$$(b+a) + (b-a) = 2\overline{X} + 2\sqrt{3\hat{S}^2}$$$$2b = 2\left(\overline{X} + \sqrt{3\hat{S}^2}\right) \Rightarrow b = \overline{X} + \sqrt{3\hat{S}^2}$$Restando la ecuación (2) de la ecuación (1) eliminamos $b$:$$(b+a) - (b-a) = 2\overline{X} - 2\sqrt{3\hat{S}^2}$$$$2a = 2\left(\overline{X} - \sqrt{3\hat{S}^2}\right) \Rightarrow a = \overline{X} - \sqrt{3\hat{S}^2}$$Finalmente, los estimadores que dependen de $X_i$ son:$\hat{a} = \overline{X} - \sqrt{\frac{3}{n}\sum(X_{i} - \overline{X})^2}$

 $\hat{b} = \overline{X} + \sqrt{\frac{3}{n}\sum(X_{i} - \overline{X})^2}$

 **Conclusión :** El método de momentos nos revela que para estimar los límites de un intervalo uniforme, debemos tomar el centro de nuestros datos (la media) y abrir el intervalo hacia la izquierda y hacia la derecha sumando y restando un margen de error basado en la desviación estándar de la muestra multiplicada por $\sqrt{3}$.

## 2. Método de máxima verosimilitud (MLE)
Sea $(X_{1},...,X_{n})$ una muestra aleatoria y $f(x_1,..., x_n; \theta)$ su densidad conjunta. Sea $f(x;\theta)$ la densidad con su parámetro $\theta$ desconocido.

*   **Función de verosimilitud:** $L(\theta) = \prod_{i=1}^{n}f(X_{i};\theta) = f(X_{1};\theta)...f(X_{n};\theta)$.
*   **Log-verosimilitud:** $\log(L(\theta)) = \log\left(\prod_{i=1}^{n}f(X_{i};\theta)\right) = \sum_{i=1}^{n}\log(f(X_{i};\theta))$.

Calculamos la derivada con respecto al parámetro a estimar y evaluamos en cero: $\frac{d \log(L(\theta))}{d\theta} = 0$. La solución es el estimador MLE.

$\color{blue}{\text{Ejercicio estimador por MLE v.a Bernoulli.}}$

Sea $(X_{1},...,X_{n})$ una muestra aleatoria $\text{Ber}(p)$. Hallar $p$ por MLE.
La función de densidad es $f(x;p) = p^{x}(1-p)^{1-x}$.

**Paso 1:**
$$ L(p) = \prod_{i=1}^{n}f(X_{i};p) = p^{X_{1}}(1-p)^{1-X_{1}} \cdots p^{X_{n}}(1-p)^{1-X_{n}} $$
$$ = p^{\sum X_{i}}(1-p)^{\sum(1-X_{i})} = p^{\sum X_{i}}(1-p)^{n-\sum X_{i}} $$

**Paso 2:**
$$ \ln(L(p)) = \ln\left(p^{\sum X_{i}}(1-p)^{n-\sum X_{i}}\right) $$
$$ = \ln(p^{\sum X_{i}}) + \ln((1-p)^{n-\sum X_{i}}) $$
$$ = \sum X_{i} \ln(p) + (n-\sum X_{i})\ln(1-p) $$

**Paso 3:**
Derivamos e igualamos a cero:
$$ \frac{d \ln(L(p))}{dp} = \frac{d(\sum X_{i} \ln(p))}{dp} + \frac{d((n-\sum X_{i})\ln(1-p))}{dp} $$
$$ = \frac{\sum X_{i}}{p} - \frac{(n-\sum X_{i})}{1-p} = 0 $$

Despejando $p$, se tiene:
$$ (1-p)\sum X_{i} - p(n-\sum X_{i}) = 0 $$
$$ \sum X_{i} - p\sum X_{i} - np + p\sum X_{i} = \sum X_{i} - np = 0 $$
$$ p = \frac{\sum X_{i}}{n} = \overline{X} $$

Por lo tanto, $\hat{p} = \overline{X}$ es el estimador MLE para $p$.

**Conclusión :** El método de máxima verosimilitud demuestra matemáticamente que la estimación más lógica y probable para la probabilidad de éxito $p$ es, de hecho, la más intuitiva: simplemente calcular la proporción de éxitos observados en nuestra muestra (la media muestral).

$\color{blue}{\text{Ejercicio estimador por MLE v.a Poisson.}}$

Sea $(X_{1},...,X_{n})$ una muestra aleatoria $\text{Poisson}(\lambda)$. Hallar $\lambda$ por MLE.

La función de probabilidad para modelar eventos en un intervalo determinado es $f(x;\lambda) = \frac{e^{-\lambda}\lambda^{x}}{x!}$.

Paso 1:$$L(\lambda) = \prod_{i=1}^{n}f(X_{i};\lambda) = \frac{e^{-\lambda}\lambda^{X_{1}}}{X_{1}!} \cdots \frac{e^{-\lambda}\lambda^{X_{n}}}{X_{n}!}$$Agrupando los términos del numerador mediante propiedades de los exponentes y dejando el denominador como una productoria:$$= \frac{e^{-n\lambda}\lambda^{\sum X_{i}}}{\prod_{i=1}^{n} X_{i}!}$$Paso 2:$$\ln(L(\lambda)) = \ln\left(\frac{e^{-n\lambda}\lambda^{\sum X_{i}}}{\prod_{i=1}^{n} X_{i}!}\right)$$Aplicando las leyes de los logaritmos para productos y divisiones:$$= \ln(e^{-n\lambda}) + \ln\left(\lambda^{\sum X_{i}}\right) - \ln\left(\prod_{i=1}^{n} X_{i}!\right)$$Simplificando:$$= -n\lambda + \left(\sum X_{i}\right)\ln(\lambda) - \ln\left(\prod_{i=1}^{n} X_{i}!\right)$$Paso 3:Derivamos con respecto a $\lambda$ e igualamos a cero. Es importante notar que el último término, al no contener $\lambda$, se vuelve cero al derivar:$$\frac{d \ln(L(\lambda))}{d\lambda} = -n + \frac{\sum X_{i}}{\lambda} - 0 = 0$$Despejando $\lambda$, se tiene:$$\frac{\sum X_{i}}{\lambda} = n$$$$\lambda = \frac{\sum X_{i}}{n} = \overline{X}$$

Por lo tanto, $\hat{\lambda} = \overline{X}$ es el estimador MLE para $\lambda$.

**Conclusión :** Para la distribución de Poisson, que se utiliza para modelar el número de veces que ocurre un evento en un lapso de tiempo o espacio, el estimador de máxima verosimilitud nos confirma que la mejor manera de estimar su tasa poblacional teórica $\lambda$ es calculando directamente el promedio empírico de las ocurrencias observadas en nuestros datos.

$\color{blue}{\text{Ejercicio estimador por MLE v.a Binomial (Simulación computacional)}}$

Lanzamos una moneda 100 veces, sabiendo que se obtienen 55 soles. Encontrar el EMV para la probabilidad $p$ de obtener sol en un solo lanzamiento.

Sea
$X$: el número de soles que se obtienen al lanzar una moneda
se tiene que $X\sim \text{Bin}(100,p)$, entonces

$$ f(x;p) = \binom{100}{55}p^{55}(1-p)^{100-55}$$

In [17]:
from scipy.stats import binom
import numpy as np

# Vamos a utilizar la libreria Sympy para matemática símbolica
import sympy

# importa la variable símbolica x
from sympy.abc import x

# Vamos a definir a p como una variable símbolica positiva: 0<p<1
p = sympy.symbols('p', positive= True)

# Definimos la función de verosimilitus
# se puede omitir la constante
f = p**55 * (1-p)**45

# Ahora derivamos f con respecto a p
df = sympy.diff(f,p)

# Hay que igualar la derivada a cero y despejar a p
# df=0 para encontrar p
phat = sympy.solve(df,p)[0] # [1] es la segunda solucion y [0] es la primera

print("El estimador de máxima verosimilitud es", phat)

El estimador de máxima verosimilitud es 11/20


**Conclusión :** El método de máxima verosimilitud confirma matemáticamente lo que nos dice la intuición: la mejor forma de estimar la probabilidad de éxito $p$ es calculando la proporción directa de nuestro experimento. Al obtener 55 "soles" en 100 lanzamientos, el cálculo en Python nos demuestra que el estimador más óptimo y probable es exactamente $\hat{p} = 0.55$ (o 55%).

$\color{blue}{\text{Ejercicio estimador por MLE v.a Bernoulli (Simulación computacional).}}$

Se genera una muestra aleatoria de $100$ observaciones a partir de un ensayo de Bernoulli con una probabilidad de éxito teórica $p=0.7$. Obtener el estimador por maxima verosimilitud.

In [18]:
# n=1 : un ensayo de Bernoulli
# p : es la probabilidad de éxito
# size : genera 100 observaciones
r = binom.rvs(n=1, p=0.7, size = 100)

# Vamos a calcular EMV

# Definimos la función de verosimilitud
f = p**x * (1-p)**(1-x)

# Calculamos el producto de todos los elementos
# subs(x,i) sustituye la variable símbolica x por una valor concreto de la muestra
J = np.prod([f.subs(x,i) for i in r])

# Calculamos el logJ de manera expandida
# sympy.log(J) = LogJ
logJ = sympy.expand_log(sympy.log(J))

# Calculamos da derivada e igualamos a cero
phat = sympy.solve(sympy.diff(logJ,p),p)[0]
print("El estimador de máxima verosimilitud es", phat)

# Podemos verificar  las soluciones con:
sol = sympy.solve(sympy.diff(logJ,p),p)
print("Verificamos que sol :",sol)

El estimador de máxima verosimilitud es 67/100
Verificamos que sol : [67/100]


**Conclusion :** el programa nos confirma que la mejor forma de estimar $p$ es, simplemente, calculando el promedio de los éxitos obtenidos en nuestra simulación.

$\color{blue}{\text{Ejercicio estimador por MLE v.a Exponencial (Simulacion Computacional)}}$

El tiempo de retardo para pacientes con enfermedad coronaria potencial se define como la duración entre el comienzo de un síntoma agudo cardiovascular y la llegada a la unidad de emergencia.

Supongamos que el tiempo de retardo tiene una distribución exponencial de parámetro $\theta$.

Se tienen los siguientes registros de retardo (en minutos) en la clinica para los primeros 20 pacientes:

$$ 525, \ 719, \ 2880, \ 150, \ 30, \ 251, \ 45, \ 858, \ 15, \ 47, \ 90, \ 56, \ 68, \ 6, \ 189, \ 180, \ 60, \ 60, \ 294, \ 747 $$

- Encontrar el estimador por el método de momentos para la media
- Encontrar el valor del estimador con los datos.

Se tiene que
$$ X : \text{tiempo de retardo (en minutos)}$$
y que $X\sim\text{Exponencial}(\theta)$.

Recordemos que $\mathbb{E}[X]=1/\theta$ y y que $\text{Var}(X)= 1 /\theta^{2}$. Además el primer momento muestra es $\bar{X}$ y despejando a $\theta$ se tiene que
$$\hat{\theta} = 1/\bar{X}$$

In [19]:
import numpy as np

# Ahora con los datos observados
datos = np.array([525, 719, 2880, 150, 30, 251, 45, 858, 15, 47, 90, 56, 68, 6, 189, 180, 60, 60, 294, 747])

# El tamaño de la muestra es
n = len(datos)

# Es estimador para la media muestral es
media_muestral = np.mean(datos)
print("La media muestral es", media_muestral)


La media muestral es 363.5


**Conclusion :** Se tiene que $\hat{\theta} = \frac{1}{\bar{X}} \approx 0.00275$ por minuto.

Por lo tanto, estimamos que el tiempo promedio de retardo entre el inicio de los sintomas cardiovasculares y la llegada a la unidad de emergencia es de aproximadamente 363.5 minutos.


# Intervalos de confianza

Un intervalo de confianza es un rango de valores que con cierta probabilidad (nivel de confianza) contiene al verdadero parámetro poblacional que queremos estimar.

Sea $X_1,\dots, X_n$ una muestra aleatoria con $X\sim\text{N}(\mu,\sigma^2)$, entonces
$$(num_1 < \mu < num2)$$

En los intervalos de confianza, tenemos el error estándar (SE) el cual mide cuanto varía un estimador.
 - si conocemos $\sigma$, entonces: $SE = \sigma/\sqrt{n}$
 - si NO conocemos $\sigma$, utilizamos $s$ y entonces: $SE = s/\sqrt{n}$

NOTA: a mayor tamaño de la muestra se tiene que el SE será menor y por lo tanto el intervalo de confianza será más preciso.

**Def** Si $X_1,\dots, X_n$ es una muestra aleatoria con densidad $f(x;\theta)$ y sean $T_1=t_1(X_1,\dots, X_n)$ y $T_2=t_2(X_1,\dots, X_n)$ dos estadísticos talque $T_1 \le T_2$ entonces
$$\mathbb{P}(T_1 \le \theta \le T_2) = \gamma$$
Al intervalo $(T_1,T_2)$ se le llama intervalo del $\gamma\times 100\%$ de confianza.


**1. Intervalo de confianza para la media de una población normal**
Sea $X_1,\dots, X_n$ una muestra aleatoria con $X\sim\text{N}(\mu,\sigma^2)$, entonces
 - $\bar{X_i} \sim \mathcal{N}(\mu, \sigma^2/n)$
 - $Z = \frac{\bar{X_i}-\mu}{\sigma/n} \sim \mathcal{N}(0,1)$

 **$\sigma$ conocida:**

Se tiene que el intervalo del $(1-\alpha)\times 100\%$ de confianza para $\mu$ es:
$$ \bar{X}\pm z_{\alpha/2}\frac{\sigma}{\sqrt{n}}$$
donde
 - $z_{\alpha/2}\frac{\sigma}{\sqrt{n}}$ es el margen de error
 - $z_{\alpha/2}$ es el valor crítico

**Ejemplo 1** Para tratar de estimar la media de consumo por cliente en un restaurante, se reunieron datos de una muestra de tamaño $49$ clientes durante tres semanas.

**a)** Supongamos que la desviación estándar poblacional es de $\$2.5$. ¿Cuál es el error estandar de la media?

**b)** Con un nivel de confianza del $95\%$, ¿cuál es el margen de error?

**c)** Si la media muestral es de $\$22.60$, cual es el intervalo de confianza del $95\%$ para la media poblacional.

In [20]:
import numpy as np
import matplotlib.pyplot as plt
import scipy.stats as st

# Datos:
n = 49 # número de clientes
sigma = 2.5
media_muestral = 22.60
confianza = 0.95

# a) SE
error_est = sigma / np.sqrt(n)
print('El error estándar es:', error_est)

# b) 1-confianza = 0.95 -> 0.025 en cada cola -> (1-confianza)/2
z = st.norm.ppf(1-(1-confianza)/2)
margen_error = z*error_est
print('El margen de error es:', margen_error)

# c) Intervalo de confianza
limite_inf = media_muestral - margen_error
limite_sup = media_muestral + margen_error
print(f'El intervalo del {confianza*100}% es: ({limite_inf:.2f},{limite_sup:.2f})')

El error estándar es: 0.35714285714285715
El margen de error es: 0.6999871373357336
El intervalo del 95.0% es: (21.90,23.30)


**Conclusión :** Aunque nuestra muestra de clientes gastó un promedio exacto de 22.60 dólares, al aplicarle el margen de error podemos afirmar con un 95% de seguridad que el consumo promedio real de todos los clientes del restaurante siempre estará entre 21.90 y 23.30 dólares.

$\sigma$ desconocida:

Se tiene que el intervalo del $(1-\alpha)\times 100\%$ de confianza para $\mu$ es:
$$ \bar{X}\pm t_{\alpha/2}\frac{\sigma}{\sqrt{n}}$$
donde
 - $t_{\alpha/2}\frac{\sigma}{\sqrt{n}}$ es el margen de error
 - $t_{\alpha/2}$ es el valor crítico

**Ejemplo 2**  Supongamos que tenemos los siguientes datos de una muestra aleatoria de 10 observaciones:
$$45,55,67,68,45,79,98,87,84,82$$

Calcular un intervalo de confianza para la media.

In [21]:
# Datos
datos = [45,55,67,68,45,79,98,87,84,82] # n<= 30
n = len(datos)
confianza = 0.99
gl = n-1

media_muestral = np.mean(datos)
error_est = st.sem(datos) # eror estandar muestral

# Vamos a utilizar la función
# scipy.stats.t.interval(confidence, df, loc, scale)
# confidence = confianza
# df = degree freedom (gl)
# loc = media_muestral
# scale = error_est

intervalo = st.t.interval(confianza, gl, media_muestral, error_est)
print(f'El intervalo de confianza del {confianza*100}%: ({intervalo[0]:.2f},{intervalo[1]:.2f})')

El intervalo de confianza del 99.0%: (52.32,89.68)


Este ejemplo nos enseña qué hacer cuando tenemos muy pocos datos (solo 10 observaciones) y desconocemos la variabilidad real. Como la muestra es tan pequeña, el cálculo ya no usa la distribución normal, sino la"t de Student" . Esta funcion ajusta y hace nuestro margen de error un poco más ancho para compensar la falta de datos, permitiéndonos estimar con un 99% de seguridad que el promedio general esta entre 52.32 y 89.68.

**2. Intervalo de confianza para la varianza de una población normal**

**$\mu$ desconocida:**

Se tiene que el intervalo del $(1-\alpha)\times 100\%$ de confianza para $\sigma^2$ es:
$$ \left( \frac{(n-1)s^2}{\chi^{2}_{1-\alpha/2}}, \frac{(n-1)s^2}{\chi^{2}_{\alpha/2}} \right)$$

**Ejemplo 3** Los artículos de cerámica utilizados sobre velas electricas se rompen con diferentes presiones. Supongamos que los datos provienen de una distribución normal. La resistencia a la ruptura fue medida en una muestra de 100 artículos y el promedio fue de $1750$ con una desviación estándar de $315.8$.

**a)** Estimar con un nivel de confianza del $90\%$ a la media poblacional de la presión de la ruptura.

**b)** Estimar con un nivel de confianza del $90\%$ a la varianza poblacional.

In [22]:
import numpy as np
import scipy.stats as st

# Datos iniciales del problema
n = 100                  # Tamaño de la muestra (artículos de cerámica evaluados)
confianza = 0.90         # Nivel de confianza deseado (90%)
media_muestral = 1750    # Promedio empírico de resistencia a la ruptura
desv_est = 315.8         # Desviación estándar observada en la muestra

# a) Estimar la MEDIA poblacional de la presión de ruptura

# Calculamos el error estándar de la media (SE = sigma / raiz(n))
error_est = desv_est / np.sqrt(n)

# Como n >= 30, por el Teorema del Límite Central podemos asumir normalidad.
# Utilizamos la función de intervalo de la distribución normal en SciPy:
# loc = estimador puntual (la media), scale = el margen de error estándar
intervalo_media = st.norm.interval(confianza, loc=media_muestral, scale=error_est)

print("INTERVALO PARA LA MEDIA")
print(f'El intervalo de confianza del {confianza*100}% para la MEDIA es: ({intervalo_media[0]:.2f}, {intervalo_media[1]:.2f})\n')


# b) Estimar la VARIANZA poblacional
gl = n - 1               # Grados de libertad (n-1)
alpha = 1 - confianza    # Nivel de significancia (en este caso 0.10)
s2 = desv_est**2         # Varianza muestral (desviación estándar elevada al cuadrado)

# Forma 1. Calculando los cuartiles de la chi cuadrada manualmente
# .ppf() calcula el valor crítico (la inversa de la distribución acumulada)
chi_inf = st.chi2.ppf(alpha/2, gl)       # Límite de la cola izquierda
chi_sup = st.chi2.ppf(1 - alpha/2, gl)   # Límite de la cola derecha

# Aplicamos la fórmula del intervalo: [(n-1)*s^2 / chi_sup, (n-1)*s^2 / chi_inf]
# OLos límites se "cruzan". Para obtener el límite inferior del intervalo,
# tenemos que dividir entre el valor más grande de Chi-cuadrada (chi_sup), y viceversa.
int_var_f1 = ((gl * s2) / chi_sup, (gl * s2) / chi_inf)

print("INTERVALO PARA LA VARIANZA")
print(f'Forma 1 - El intervalo de confianza del {confianza*100}% para la VARIANZA es: ({int_var_f1[0]:.2f}, {int_var_f1[1]:.2f})')

# Forma 2. Usando la función integrada de SciPy
# st.chi2.interval nos da exactamente los mismos cuantiles chi_inf y chi_sup
chi2_inf, chi2_sup = st.chi2.interval(confianza, gl)
int_var_f2 = ((gl * s2) / chi2_sup, (gl * s2) / chi2_inf)

print(f'Forma 2 - El intervalo de confianza del {confianza*100}% para la VARIANZA es: ({int_var_f2[0]:.2f}, {int_var_f2[1]:.2f})')

# Calculamos el intervalo para la DESVIACIÓN ESTÁNDAR sacando la raíz cuadrada
# Esto es mucho más útil para interpretar el resultado final en la vida real.
int_desv = (np.sqrt(int_var_f1[0]), np.sqrt(int_var_f1[1]))
print(f'\nIntervalo para la DESVIACIÓN ESTÁNDAR: ({int_desv[0]:.2f}, {int_desv[1]:.2f})')

INTERVALO PARA LA MEDIA
El intervalo de confianza del 90.0% para la MEDIA es: (1698.06, 1801.94)

INTERVALO PARA LA VARIANZA
Forma 1 - El intervalo de confianza del 90.0% para la VARIANZA es: (80123.49, 128146.72)
Forma 2 - El intervalo de confianza del 90.0% para la VARIANZA es: (80123.49, 128146.72)

Intervalo para la DESVIACIÓN ESTÁNDAR: (283.06, 357.98)


**Conclusión :**

1.Para el intervalo de la media: podemos decir que con un $90\%$ de nivel de confianza, podemos afirmar que la verdadera presión de ruptura promedio de todas las piezas de cerámica fabricadas en esa línea de producción se encuentra entre $1698.05$ y $1801.95$.

2.Para el intervalo de la varianza: la varianza nos dio un intervalo con valores enormes, lo cual es normal porque está en "presión al cuadrado". Al sacarle la raíz cuadrada (Desviación Estándar), encontramos que con un $90\%$ de confianza, la variacion típica de la resistencia entre las diferentes piezas está entre $283$ y $357$.

#Conclusión

Es este notebook nos concentramos en los estimadores de una variable aleatoria que seria una funcion para aproximar el mejor parametro que pueda explicar nuestro modelo, vimos que existen diferentes propiedades las cuales deben de cumplir para ser buenos estimadore y revisamos las 2 formas de calcularlos, por el metodo de momentos y por maxima verosimilitud este ultimo un poco mas comodo si tenemos python o algun lenguaje de programacion a la mano y por ultimo revisamos los intervalos de confianza de un parametro, los cuales son escenciales para definir a travez de un rango los posibles valores que puede tomar una observacion.

# Bibliografia
- De la Fuente Fernández, S. (2023). Apuntes de Inferencia Estadística y Estimadores. Departamento de Matemáticas, Universidad Autónoma de Madrid.

- Materia en clase